docker/daemon_test.go

```diff
@@ -34,6 +34,49 @@ func TestLoadDaemonCliConfigWithoutOverriding(t *testing.T) {
 	}
 }
 
+func TestLoadDaemonCliConfigWithDaemonFlags(t *testing.T) {
```

Comment:
this is does not seem to be the same test as TestDaemonStartWithDaemonCommand. It seems this unit test is testing the config file, whereas TestDaemonStartWithDaemonCommand is testing flags on the daemon cli.